optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/config/ProblemBenchmarksConfig.java

```diff
@@ -178,22 +179,35 @@ private ProblemBenchmarkResult buildProblemBenchmark(PlannerBenchmarkResult plan
         return problemBenchmarkResult;
     }
 
-    private void buildSingleBenchmark(
-            SolverBenchmarkResult solverBenchmarkResult, ProblemBenchmarkResult problemBenchmarkResult) {
+    private void buildSingleBenchmark(SolverBenchmarkResult solverBenchmarkResult,
+            ProblemBenchmarkResult problemBenchmarkResult) {
         SingleBenchmarkResult singleBenchmarkResult = new SingleBenchmarkResult(solverBenchmarkResult, problemBenchmarkResult);
-        List<PureSingleStatistic> pureSingleStatisticList = new ArrayList<PureSingleStatistic>(
-                singleStatisticTypeList == null ? 0 : singleStatisticTypeList.size());
-        if (singleStatisticTypeList != null) {
-            for (SingleStatisticType singleStatisticType : singleStatisticTypeList) {
-                pureSingleStatisticList.add(singleStatisticType.buildPureSingleStatistic(singleBenchmarkResult));
+        buildSubSingleBenchmarks(singleBenchmarkResult, solverBenchmarkResult.getSubSingleCount());
+        for (SubSingleBenchmarkResult subSingleBenchmarkResult : singleBenchmarkResult.getSubSingleBenchmarkResultList()) {
+            subSingleBenchmarkResult.setPureSubSingleStatisticList(new ArrayList<PureSubSingleStatistic>(
+                    subSingleStatisticTypeList == null ? 0 : subSingleStatisticTypeList.size()));
+        }
+        if (subSingleStatisticTypeList != null) {
+            for (SubSingleStatisticType subSingleStatisticType : subSingleStatisticTypeList) {
+                for (SubSingleBenchmarkResult subSingleBenchmarkResult : singleBenchmarkResult.getSubSingleBenchmarkResultList()) {
+                    subSingleBenchmarkResult.getPureSubSingleStatisticList().add(subSingleStatisticType.buildPureSubSingleStatistic(subSingleBenchmarkResult));
+                }
             }
         }
-        singleBenchmarkResult.setPureSingleStatisticList(pureSingleStatisticList);
-        singleBenchmarkResult.initSingleStatisticMap();
+        singleBenchmarkResult.initSubSingleStatisticMaps();
         solverBenchmarkResult.getSingleBenchmarkResultList().add(singleBenchmarkResult);
         problemBenchmarkResult.getSingleBenchmarkResultList().add(singleBenchmarkResult);
     }
 
+    private void buildSubSingleBenchmarks(SingleBenchmarkResult parent, int count) {
```

Comment:
nitpicking: I might name the count param subSingleCount to avoid any confusion (and to make it clear it indirectly comes from the <subSingleCount> element in the configs)